pkg/api/meta/restmapper.go

```diff
@@ -206,23 +206,32 @@ func (m *DefaultRESTMapper) ResourcesFor(resource unversioned.GroupVersionResour
 
 	case hasGroup:
 		requestedGroupResource := resource.GroupResource()
-		for currResource := range m.pluralToSingular {
-			if currResource.GroupResource() == requestedGroupResource {
-				ret = append(ret, currResource)
+		for plural, singular := range m.pluralToSingular {
+			if singular.GroupResource() == requestedGroupResource {
+				ret = append(ret, plural)
+			}
+			if plural.GroupResource() == requestedGroupResource {
+				ret = append(ret, plural)
 			}
 		}
 
 	case hasVersion:
-		for currResource := range m.pluralToSingular {
-			if currResource.Version == resource.Version && currResource.Resource == resource.Resource {
-				ret = append(ret, currResource)
+		for plural, singular := range m.pluralToSingular {
```

Comment:
> I am skeptical that this is a useful case. If I know I'm looking for a "v1" "Pod" and my choices are the main api or the metrics group, one of those is super wrong.

As to the more general question of "who chooses winning groups", `ResourcesFor` is made for the purpose of finding all known matches and putting them in a preferred ordering (note the sorting happening below).  Choosing a single winner is built on top of this logic, but it doesn't remove the necessity of finding all the matches before selecting one.